MSVC Compiler Error C4737 by async_simple::coro::Lazy

### Search before asking

- [√ ] I searched the [issues](https://github.com/alibaba/async_simple/issues) and found no similar issues.


### What happened + What you expected to happen

A simple function cause compile error in MSVC Release.

```cpp
#include "async_simple/coro/Lazy.h"
#include "async_simple/coro/SyncAwait.h"

async_simple::coro::Lazy<int> test(int val) { co_return val; }

int main() { return async_simple::coro::syncAwait(test(0)); }
```

error info:

```
main.cpp(19): error C4737: 无法执行所需尾调用。性能可能会降级。
```

### Reproduction way

VS 2022 17.3.4 Release

### Anything else

Add `/EHa` can fix this problem. It seems a bug by optimization in MSVC Release.


### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

--- a/README.md
+++ b/README.md
@@ -113,7 +113,9 @@ cmake .. && sudo make install
 
 Required Compiler: clang (>= 13.0.0) or gcc (>= 10.3)
 
-Note that we need to add `-Wno-maybe-uninitialized` option when we use gcc12 due to a false positive diagnostic message by gcc12.
+Note that we need to add `-Wno-maybe-uninitialized` option when we use gcc12 due to a false positive diagnostic message by gcc12
+
+If you meet any problem about MSVC Compiler Error C4737. Try to add option /EHa to fix the problem.
 
 ## Demo example dependency
 

--- a/README_CN.md
+++ b/README_CN.md
@@ -19,6 +19,8 @@ async\_simple 涉及 C++20 协程，对编译器版本有较高要求。需要 c
 
 注意当使用 gcc12 时需要加上 `-Wno-maybe-uninitialized` 选项因为 gcc12 的一个误报。详情可见 https://github.com/alibaba/async_simple/issues/234。
 
+注意，如果使用msvc时遇到了C4737错误，请加上选项/EHa。
+
 ## Debian/Ubuntu系列
 
 - 安装clang11及其以上版本。安装方法见：[APT Packages](https://apt.llvm.org/)

